post: gsoc-final-report (#7)
Grillo-0/blog: _posts/2023-08-13-gsoc-2023-final-report.md

---
layout: post
title: "GSoC 2023 Final Report"
date: 2023-08-13 18:51:37 -0300
category: kernel developing
tags: GSoC
---

The GSoC journey is coming to a close. In just over 100 days, I gained

> more experience in open-source development than I could ever imagine

Reviewer: ```suggestion
more experience, in open-source development than I could ever imagine
```

:)

This is gonna be a report of all my contributions and interaction with
the community.

# GSoC contributions

## Linux Kernel - VKMS

VKMS is a software-only model of a KMS driver that is useful for
testing and running X (or similar) on headless machines.

This was, unexpectedly, a big part of my GSoC. I learned a lot about
color formats and how a graphics driver works. Currently, only one
piece of my work was upstreamed, the rest needs more work and was
postponed in favor of the primary project goal.

| Patch | Status |
|-------------------------------------------------------------------------------------------------------------------------|--------------|
| [drm/vkms: Add support to 1D gamma LUT](https://lore.kernel.org/all/20230621194121.184552-1-arthurgrillo@riseup.net/#r) | **Accepted** |

For more information go check my [blogpost]({{site.baseurl}}{%
link _posts/2023-06-19-nv12-vkms.md %}) about the whole process.

## IGT

IGT GPU Tools is a collection of tools for the development and testing
of DRM drivers. While working on VKMS I used heavily the IGT framework
for testing, in one occasion a bug made the test that I was using

> stopped working on the VKMS, so a submitted a patch to fix that.

Reviewer: ```suggestion
stop working on the VKMS, so a submitted a patch to fix that.
```

| [Re: [PATCH] drm/tests: Remove CONFIG_DRM_FBDEV_EMULATION on .kunitconfig](https://lore.kernel.org/r/bef940cb-b079-72ce-692c-3b6c6d283265@riseup.net/)                 |
| [Re: [PATCH] drm/tests: Remove CONFIG_DRM_FBDEV_EMULATION on .kunitconfig](https://lore.kernel.org/r/74e5cc8c-a5f8-4c0b-c699-4234e539203f@riseup.net/)                 |
| [Re: [PATCH] drm/tests: Alloc drm_device on drm_exec tests](https://lore.kernel.org/r/591114cb-88f7-0a43-f2ba-8ab5836571c9@riseup.net/)                                |
| [Re: [PATCH] drm: Drop select FRAMEBUFFER_CONSOLE for DRM_FBDEV_EMULATION](https://lore.kernel.org/r/7950bcea-0f15-da2e-e4f7-4bfa474a420f@riseup.net/)                 |
| [Re: [PATCH -next 6/7] drm/format-helper: Remove unnecessary NULL values](https://lore.kernel.org/r/28268a1b-090c-237a-79dd-ca58de712a1e@riseup.net/)                  |
| [Re: [PATCH 6/6] drm/format-helper: Add KUnit tests for drm_fb_memcpy()](https://lore.kernel.org/r/4cee9366-57c7-ec70-b547-072cc0a2d157@riseup.net/)                   |
| [Re: [PATCH v2 6/6] drm/tests: Add KUnit tests for drm_fb_memcpy()](https://lore.kernel.org/r/3dffe17d-4c6d-4cb7-2c22-c6d747e98bb0@riseup.net/)                        |

> Moreover, I use a thunderbird addon to make the diff properly

Reviewer: ```suggestion
Moreover, I use a Thunderbird addon to make the diff properly
```

> highlyted. Whwn I was thinkering with configuration, I noticed that

Reviewer: ```suggestion
highlighted. When I was tinkering with the configuration, I noticed that
```

> the CSS of the coniuguration menu was wrong, so it made the use

Reviewer: ```suggestion
the CSS of the configuration menu was wrong, so it made the user
```

experience pretty bad.

I sent a [patch](https://github.com/Qeole/colorediffs/pull/116) fixing

> that to the the mainteiner of the addon, this patch generated a

Reviewer: ```suggestion
that to the maintainer of the addon, this patch generated a
```

> discussion that made a hole change of the CSS file due to thunderbird

Reviewer: ```suggestion
discussion that made a whole change in the CSS file due to Thunderbird
```

updates.

# Acknowledgments

I'd like to thank my mentors, André “Tony” Almeida, Tales
L. Aparecida, and Maíra Canal. Their support and expertise were
invaluable throughout this journey.

> Moreover, I thank the X.Org foundation for allowing me to participate

Reviewer: ```suggestion
Moreover, I thank the X.Org Foundation for allowing me to participate
```

---
layout: post
title: "GSoC 2023 Final Report"
date: 2023-08-24 00:00:00 -0300
category: kernel developing
tags: GSoC
---

The GSoC journey is coming to a close. In just over 100 days, I gained

> more experience, in open-source development than I could ever imagine

Reviewer 2: ```suggestion
more experience in open-source development than I could ever imagine
```

in this period.

> Prior to GSoC, I was not accustomed to regularly submitting patches to

Reviewer 2: ```suggestion
Prior to GSoC, I was not used to regularly submitting patches to
```

> Prior to GSoC, I was not accustomed to regularly submitting patches to

Reviewer 2: ```suggestion
Prior to GSoC, I was not accustomed to regularly submit patches to
```

> the mailing lists. Now, I've sent many patches and revisions.  I think

Reviewer 2: ```suggestion
the mailing lists. Now, I've sent many patches and revisions.  I believe
```

my interaction with the community will only grow. I learned so much
about the tools and workflow of kernel development.

After this experience, I'm more than certain that I want to make this a
job, contributing to open-source is fun, so why not make this a living
:)

# Goals

> The main goal was to increase the code coverage on the DRM core helper

Reviewer 2: ```suggestion
The main goal of the project was to increase the code coverage on the DRM core helper
```

| [drm/tests: Add multi-plane support to conversion_buf_size()](https://lore.kernel.org/r/20230814-gsoc-drm-format-test-v2-v3-5-bd3e9f9bc2fb@riseup.net/) | **On Review** |
| [drm/tests: Add KUnit tests for drm_fb_memcpy()](https://lore.kernel.org/r/20230814-gsoc-drm-format-test-v2-v3-6-bd3e9f9bc2fb@riseup.net/)              | **On Review** |

# Challenges Faced

I think the most difficult task was describing my work. Either on blog
posts or in the commit messages, it takes a lot of work to write what
you've done concisely and clearly. With time you get the way of
things, but I think I can improve on this subject.

Moreover, many times I had to debug some problems. I already knew how
to use GDB, but using it in the kernel is a little more
cumbersome. After searching, reading the documentation, and getting
tips from my mentors, I got it working.

On the VKMS, I had to create new features, this requires a lot of
thought. I made a lot of diagrams in my head to understand how the
color formats would be displayed in memory, and yet most of my work
hasn't seen the light of day XD.

> # What is Left

Reviewer 2: ```suggestion
# What is left to do 
```
